Wrong type for Display.float_format

**Describe the bug**
`Display.float_format` is annotated as `str | None`. However, the correct type should be `Callable[[float], str]`. From [the documentation](https://pandas.pydata.org/docs/user_guide/options.html):
```
display.float_format : callable
    The callable should accept a floating point number and return
    a string with the desired format of the number. This is used
    in some places like SeriesFormatter.
    See formats.format.EngFormatter for an example.
    [default: None] [currently: None]
```

**To Reproduce**
MRE:
```
import pandas as pd

pd.options.display.float_format = '{:,.2f}'.format

df = pd.DataFrame([1.1, 2.2, 3.3, 4.4],
                  index=['a', 'b', 'c', 'd'],
                  columns=['value'])

print(df)
```
Mypy output:
```
$ mypy mre.py  
foo.py:3: error: Incompatible types in assignment (expression has type "Callable[[VarArg(object), KwArg(object)], str]", variable has type "Optional[str]")  [assignment]
Found 1 error in 1 file (checked 1 source file)
```


**Please complete the following information:**
 - OS: Arch Linux
 - OS Version: Linux 5.15.72
 - Python 3.10.7
 - Mypy and Pyright
 - `pandas-stubs` version 1.5.3.230214

--- a/pandas-stubs/_config/config.pyi
+++ b/pandas-stubs/_config/config.pyi
@@ -1,4 +1,7 @@
-from collections.abc import Iterable
+from collections.abc import (
+    Callable,
+    Iterable,
+)
 from contextlib import ContextDecorator
 from typing import (
     Any,
@@ -51,7 +54,7 @@ class Display(DictWrapper):
     date_yearfirst: bool
     encoding: str
     expand_frame_repr: bool
-    float_format: str | None
+    float_format: Callable[[float], str] | None
     html: DisplayHTML
     large_repr: str
     latex: DisplayLaTeX

--- a/tests/test_config.py
+++ b/tests/test_config.py
@@ -1,6 +1,8 @@
 from typing import (
     TYPE_CHECKING,
     Any,
+    Callable,
+    Optional,
 )
 
 import pandas as pd
@@ -36,3 +38,13 @@ def test_specific_option():
     check(assert_type(pd.options.plotting.backend, str), str)
     # Just check assignment
     pd.options.plotting.backend = "matplotlib"
+
+
+def test_display_float_format():
+    check(
+        assert_type(pd.options.display.float_format, Optional[Callable[[float], str]]),
+        type(None),
+    )
+    formatter = "{,.2f}".format
+    with pd.option_context("display.float_format", formatter):
+        assert pd.get_option("display.float_format") == formatter

